doc/debug.md

```diff
@@ -0,0 +1,183 @@
+# Debugging the operator
+
+This document outlines the distinct debugging requirements for various components, providing detailed instructions for each to ensure effective troubleshooting and development.
+
+## TLDR
+
+### CSI-Driver-server
+
+- **Run CSI driver executables on the node** for file system operations.
+- **Makefile commands**:
+  - `make debug/build`: Build image with Delve included.
+  - `make debug/deploy`: Install image with necessary changes to deployments.
+  - `make debug/tunnel`: Open tunnel from local machine to CSI driver pod.
+- **IntelliJ setup**: Configure "Go Remote" with `localhost` and port `40000`, and set "On disconnect" to "Leave it running".
+- **VSCode:** Add a debug configuration to "Connect to Server" with `127.0.0.1` as the host and `40000` as the port, and set "remotePath" to `github.com/Dynatrace/dynatrace-operator`.
```

Comment:
IMO: Having a "copy-pastable" json, is the most "TLDR"  😄 (you already kinda to this for the "none" TLDR sections)
```
        {
            "name": "Debug CSI server",
            "type": "go",
            "request": "attach",
            "mode": "remote",
            "remotePath": "github.com/Dynatrace/dynatrace-operator",
            "port": 40000,
            "host": "127.0.0.1"
        },
```